include/settings/settings.h

```diff
@@ -248,15 +269,57 @@ void settings_dst_register(struct settings_store *cs);
 /**
  * Parses a key to an array of elements and locate corresponding module handler.
  *
- * @param name Key in string format
- * @param name_argc Parsed number of elements.
- * @param name_argv Parsed array of elements.
+ * @param[in] name in string format
+ * @param[out] next remaining of name after matched handler
  *
  * @return settings_handler node on success, NULL on failure.
  */
-struct settings_handler *settings_parse_and_lookup(char *name, int *name_argc,
-						   char *name_argv[]);
+struct settings_handler *settings_parse_and_lookup(const char *name,
+						   const char **next);
+
+
+/*
+ * API for const name processing
+ */
+
+/**
+ * Compares the start of name with a key
+ *
+ * @param[in] name in string format
+ * @param[in] key comparison string
+ * @param[out] next pointer to remaining of name (excluding separator)
+ *
+ * @return 0: no match
+ *         1: match, next can be used to check if match is full
+ */
+int settings_name_cmp(const char *name, const char *key, const char **next);
 
+/**
+ * Splits up name in part before and after first separator ('/')
+ *
+ * @param[in] name in string format
+ * @param[out] argv part of name before first separator ('/')
+ * @param[out] next pointer to remaining of name (excluding separator)
+ *
+ * @return 0: there is no separator in name, argv contains name
+ *         1: there is a separator in name
+ */
+int settings_name_split(const char *name, char *argv, const char **next);
+
+/**
+ * Splits up name in count of items separated by separator ('/') and a list
+ * of the separated items
+ *
+ * @param[in] name in string format
+ * @param[out] name_argc items count
+ * @param[out] name_argv separated items
+ *
+ * @return 0: there is no separator in name, argv contains name
+ *         1: there is a separator in name
+ *         -ENOENT: error (this should only happen if empty string is used for
+ *                  name)
+ */
+int settings_parse_name(char *name, int *name_argc, char *name_argv[]);
```

Comment:
I do not see this function used anywhere. Note the risk of buffer overflow is not handled in the implementation.

Reply:
Yes, this function could be removed, I agree it should not be used (this will require some more work in the modules :-))

Reply:
settings_parse_name removed